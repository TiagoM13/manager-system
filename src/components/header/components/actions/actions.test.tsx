import React from 'react';

import { fireEvent, render } from '@testing-library/react';

import { HeaderActions } from '.';

describe('<HeaderActions/>', () => {
  test('should render correctly component', () => {
    const { getByText, getByTestId, asFragment, container } = render(
      <HeaderActions />,
    );

    const btnCancel = container.querySelector('.bg-red-600');
    const btnEdit = container.querySelector('.bg-sky-600');
    const iconCheck = getByTestId('icon-check');

    expect(getByText('cancelar')).toBeInTheDocument();
    expect(getByText('salvar')).toBeInTheDocument();
    expect(btnCancel).toBeInTheDocument();
    expect(btnEdit).toBeInTheDocument();
    expect(iconCheck).toBeInTheDocument();

    expect(asFragment()).toMatchSnapshot();
  });

  test('should render loading icon', () => {
    const labels = {
      cancel: 'voltar',
      saved: 'prosseguir',
    };

    const { getByText, getByTestId } = render(
      <HeaderActions buttonLabels={labels} loading />,
    );

    const iconLoading = getByTestId('icon-loading');

    expect(getByText(labels.cancel)).toBeInTheDocument();
    expect(getByText(labels.saved)).toBeInTheDocument();
    expect(iconLoading).toBeInTheDocument();
  });

  test('should call the cancel function', () => {
    const handleOnCancel = jest.fn();

    const { getByTestId } = render(<HeaderActions onCancel={handleOnCancel} />);

    const button = getByTestId('btn-cancel');

    fireEvent.click(button);

    expect(handleOnCancel).toHaveBeenCalled();
  });
});
