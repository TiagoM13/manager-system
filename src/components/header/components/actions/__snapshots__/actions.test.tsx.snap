// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HeaderActions/> should render correctly component 1`] = `
<DocumentFragment>
  <button
    class="bg-red-600 outline-red-500 hover:bg-red-500 disabled:bg-red-700 rounded-lg py-2.5 px-3 font-medium text-sm text-white flex gap-2 items-center justify-center"
    data-testid="btn-cancel"
    type="button"
  >
    <svg
      class="size-5"
      fill="currentColor"
      height="1em"
      viewBox="0 0 256 256"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M208.49,191.51a12,12,0,0,1-17,17L128,145,64.49,208.49a12,12,0,0,1-17-17L111,128,47.51,64.49a12,12,0,0,1,17-17L128,111l63.51-63.52a12,12,0,0,1,17,17L145,128Z"
      />
    </svg>
    cancelar
  </button>
  <button
    class="bg-sky-600 outline-sky-500 hover:bg-sky-500 disabled:bg-sky-700 rounded-lg py-2.5 px-3 font-medium text-sm text-white flex gap-2 items-center justify-center min-w-[100px]"
    data-testid="btn-saved"
    type="submit"
  >
    <svg
      class="size-5"
      data-testid="icon-check"
      fill="currentColor"
      height="1em"
      viewBox="0 0 256 256"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M232.49,80.49l-128,128a12,12,0,0,1-17,0l-56-56a12,12,0,1,1,17-17L96,183,215.51,63.51a12,12,0,0,1,17,17Z"
      />
    </svg>
    salvar
  </button>
</DocumentFragment>
`;
